src/server/res.rs

```diff
@@ -9,11 +9,27 @@
 //! Response factory functions.
 //!
 
-use headers::{ContentLength, HeaderMapExt};
+use headers::{ContentLength, Header, HeaderMapExt, HeaderValue, Location};
 use hyper::StatusCode;
 
 use crate::server::Response;
 
+/// Generate 304 NotModified response.
+pub fn moved_permanently(mut res: Response, src: &str) -> Response {
+    *res.status_mut() = StatusCode::MOVED_PERMANENTLY;
```

Comment:
I wonder shall we use "302 Found" instead? Since some scenarios of sfz are for temporary use, and we do not want Client to cache the redirection indefinitely.

Reply:
It's probably unlikely that a directory and a file have the same name, right?

Reply:
Yes. But the content you serve may change. We do not want client to cache 301 forever.